### PR TITLE
Error checking for config parameters, and sending errors over HTTP when in serve mode

### LINKED ISSUE #149: Error handling for config / filename
Empty body, division by 0, invalid filename

### PR DESCRIPTION
**Pull request by VFjr**
Closes #149 and #166

In the previous implementations, the code would try to run a simulation even if some vital parameters were not set. In some cases that could even cause division by 0. Now it checks that the vital parameters that have to be initialised are both initialised and not set to 0, and returns an error otherwise. 

When running in serve mode, if there is an error, it sends it as a 400 Bad Request. Here is a screenshot:

<details>
  <summary>Missing AgentHP in config, received a 400 Bad Request with the error message itself in the body </summary>

image

</details>

### CODE REVIEW

**Review by JaafarRammal — COMMENTED** [review]
Legend. Please wait for a non-infra approval before merging

File: pkg/config/configparameters.go
Comment (jzzheng22) [inline pkg/config/configparameters.go:72]:
```suggestion
	err = CalculateDependentParameters(&tempParameters)
```

File: pkg/config/configparameters.go
Comment (jzzheng22) [inline pkg/config/configparameters.go:90]:
```suggestion
	err = CalculateDependentParameters(&tempParameters)
```

File: pkg/config/configparameters.go
Comment (jzzheng22) [inline pkg/config/configparameters.go:99]:
```suggestion
func CalculateDependentParameters(parameters *ConfigParameters) error {
```

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

**Review by jzzheng22 — APPROVED** [review]
ty bby

**Pull request merged**
